site/src/api/api.ts

```diff
@@ -1738,15 +1738,20 @@ class ApiMethods {
 			name: "",
 			add_users: [userId],
 			remove_users: [],
+			display_name: null,
+			avatar_url: null,
+			quota_allowance: null,
 		});
 	};
 
 	removeMember = async (groupId: string, userId: string) => {
 		return this.patchGroup(groupId, {
 			name: "",
-			display_name: "",
 			add_users: [],
 			remove_users: [userId],
+			display_name: null,
+			avatar_url: null,
+			quota_allowance: null,
```

Comment:
I talked to @Parkreiner about this, and we have to do some overhauling of our types when it comes to `omitempty` vs `null`.

For requests, maybe `omitempty` is easier to use, but `null` forces the user to acknowledge the fields exist. This actually raised a group member bug.

This is very similar to what we did in our Golang linter for structs: https://github.com/coder/coder/pull/8403

For most objects returned by Golang, `omitempty` is incorrect, as we usually have a `*string` or some pointer to something. Go json marshals this as `null`, and the previous typescript code would treat the type as `string | undefined`, but at runtime the type is `null`. So it was just incorrect, however the truthy value is consistent.


----

tl;dr: We've been using `undefined` and `null` incorrectly in our typescript types, and it's too much to fix in this PR. I just had to make some changes to make the typescript compiler happy, even though nothing has functionally changed.

Reply:
Yeah, I think the `?:` syntax has better ergonomics in React component props, but in all other cases, having the explicit `null` case is better for making sure we don't have invalid/malformed requests

I'd rather we overhaul things so that we have to deal with explicit null cases for anything involving an API call